### PR TITLE
New addon: MyHome

### PR DESCRIPTION
**Pull request by from20020516**
# MyHome
- Automatically choose and uses a warp spell or an item.

## Command
- `//mh` OR `//warp`
- Priorities are:
    1. Warp II <me\> --BLM main or sub
    2. Warp Ring --Inventory or Wardrobe 1~4
    3. Warp Cudgel --Inventory or Wardrobe 1~4
    4. Instant Warp --Inventory

### CODE REVIEW

File: addons/MyHome/MyHome.lua [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                item_array[item.id] = item
+                item_array[item.id].bag = bags[i]
+            end
+        end
+    end
+    for index,stats in pairs(item_info) do
+        local item = item_array[stats.id]
+        local set_equip = windower.ffxi.set_equip
+        if item then
+            ext = extdata.decode(item)
```
Comment (z16) [inline addons/MyHome/MyHome.lua]: This can be made `local`.

File: addons/MyHome/MyHome.lua [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            local enchant = ext.type == 'Enchanted Equipment'
+            local recast = enchant and ext.charges_remaining > 0 and math.max(ext.next_use_time+18000-os.time(),0)
+            local usable = recast and recast == 0
+            log(stats[lang],usable and '' or recast..' sec recast.')
+            if usable or ext.type == 'General' then
+                if enchant and item.status ~= 5 then --not equipped
+                    set_equip(item.slot,stats.slot,item.bag)
+                    log_flag = true
+                    repeat --waiting cast delay
+                        coroutine.sleep(1)
```
Comment (z16) [inline addons/MyHome/MyHome.lua:68]: Why not sleep once for the entire delay, instead of looping? You can get the delay from ext data, right?

**Review by z16 — APPROVED** [review]
Everything looks good, I left two comments, but they're just minor suggestions. Let me know if you want to address them or if you want me to merge this.

File: addons/MyHome/MyHome.lua (hunk 2 above)
Comment (from20020516) [reply addons/MyHome/MyHome.lua:68]: Probably, the ffxi client detects usable enchanted gear every 'tick' (time change event)
```lua
ext.usable == delay+math.random(0,3)
```
of course I could write something like `coroutine.sleep(delay+3)`,
but users use warp high-frequently resulting in a loss of user experience.

**Comment by from20020516** [comment]
I'm ready. thank you :)

**Comment by z16** [comment]
There's a conflict with the `addons.xml` file. Can you pull the latest changes, resolve the conflict and then submit the PR again?

**Comment by from20020516** [comment]
done :)

**Pull request merged**